src/DataStructures/VectorAlgebra.hpp

```diff
@@ -0,0 +1,66 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/VectorImpl.hpp"
+#include "Utilities/Gsl.hpp"
+
+// @{
+/// \ingroup DataStructuresGroup
```

Comment:
To me this (and to a lesser extent `repeat`) feels like it belongs in a math-y documentation group, rather than the DataStructures group. I'm not sure what the consensus is on this...

Reply:
I was a bit torn on where to put it. Do you think that the functions themselves are better suited to be in the `Utilities` directory as well?

Reply:
I think I would favor putting these in the `Utilities` directory, but I am happy to be overruled :)

Reply:
Since these all depend on `Blaze` why not add them to `Utilities/PointerVector.hpp`? That's where are other code that extends Blaze lives. I also think `outer` -> `outer_product` to match `std::inner_product` naming.

Reply:
For reasons that are mysterious to me, I don't seem to be able to build if `Gsl.hpp` is included in `PointerVector.hpp` -- there seems to be some resolution trouble with a stream operator. For the time being, I'll just put `VectorAlgebra.hpp` in `Utilities` to avoid the offending includes.